### ISSUE TITLE
Varnish - X-cache MISS

### ISSUE THREAD

**Issue by mickaelprevot** [api-platform/api-platform#494]
Hello,

I've wanted to try the api-platform demo (v 2.1) and most specifically, Varnish.

Just for the fun, I have add the following code in the vcl_deliver and I realized that no data was cached. In fact, X-Cache is always to MISS. Of course, I have watched the nginx logs to confirm that varnish  always solicited nginx.

```
  if (obj.hits > 0) { 
        set resp.http.X-Cache = obj.hits + " HITS";
    } else {
        set resp.http.X-Cache = "MISS";
    }
```

Why not have implemented a "return (hash)" in the vcl_recv ?

**Comment by k911** [comment]
I've had an similar issue, and got a headache while debugging. 
My stack: PHP 7.2 fpm alpine, nginx, varnish 4, symfony 4

The problem is surprisingly caused by upgrade to symfony 3.4/4.0, exactly these lines being executed: 
https://github.com/symfony/symfony/commit/2c0dc745d63bfe227db07d4bb74adc47d585d950#diff-d29419a0d7043fce54a98005c26a82aa

Resulting in response containing these headers:
```
...
Cache-Control: max-age=0, private, must-revalidate
Vary: Accept
Vary: Authorization
X-Content-Type-Options: nosniff
X-Frame-Options: deny
Cache-Control: max-age=0, public, s-maxage=3600
...
```

as you can see 'cache-control' header is duplicated here, and first entry is obviously undesirable due to 'private', which prevent caching response...

Solution is to add this config in symfony application.
```
parameters:
   session.storage.options:
        cache_limiter: 0```

BTW. Isn't this BC break in symfony 3.4? :)

**Comment by dunglas** [comment]
@k911 are you using API Platform? By default API Platform doesn't read from the session, so the header shouldn't be sent.

**Comment by k911** [comment]
Yes, I'm using API Platform but I've used other bundles too. 

Like FOS User and Easy Admin (for development purposes, in long term should be replaced by api-platform/admin). Anyway, I use normal (session based) login on firewall with `/admin` path, and general firewall `/` is stateless, so I would expect that only `/admin` routes should use session, and send those headers.

Unless of course, I've messed up security configuration: https://github.com/knit-pk/api-v1-php/blob/develop/config/packages/security.yaml

Yes, on symfony 3.3 (with flex, but probably irrelevant), it worked just right.
